Lib/test/test_xml_etree.py

```diff
@@ -1739,6 +1740,41 @@ def __eq__(self, other):
         self.assertIsInstance(e[0].tag, str)
         self.assertEqual(e[0].tag, 'changed')
 
+    def check_expat224_utf8_bug(self, text):
+        xml = b'<a b="%s"/>' % text
+        root = ET.XML(xml)
+        self.assertEqual(root.get('b'), text.decode('utf-8'))
+
+    def test_expat224_utf8_bug(self):
+        # bpo-31170: Expat 2.2.3 had a bug in its UTF-8 decoder.
+        # Check that Expat 2.2.4 fixed the bug.
+
+        # 1 KB buffer
+        text = b'x' * 1023 + b'\xc3\xa0'
+        self.check_expat224_utf8_bug(text)
+
+        # 2 KB buffer
```

Comment:
It would be better to document these two tests as testing buffer bounds at odd and even positions.

And please use binary prefixes: KiB.

Reply:
done